clientId fix

vriken opened the pull request.

vriken committed:
clientId fix

vriken committed:
typo

vriken committed:
adding permissions

giovaniortolani commented:
Is part of: https://github.com/stape-io/data-tag/pull/42

Thanks @vriken. I'll review it. I cannot give you an ETA yet.

vriken committed:
Add cookie checks for bootstrap event handling

For returning users - If a bootstrap event comes in, immediately return 204 without processing event models, generating any IDs or setting any cookies

For new users (no _dcid): process the bootstrap event normally, set both _dcid and _bootstrap_cid cookies

vriken committed:
dont claim unecessary bootstrap events

giovaniortolani commented:
Hi @vriken, please see https://github.com/stape-io/data-tag/pull/43 and https://github.com/stape-io/data-client/pull/21. It's a simple and functional solution to the problem.

If you want to test locally, just update the **Data Tag** with the new [template.tpl](https://github.com/stape-io/data-tag/blob/fix-dcid-generation-race-condition/template.tpl) and the **Data Client** with the new [template.tpl](https://github.com/stape-io/data-client/blob/fix-dcid-generation-race-condition/template.tpl).

vriken commented:
#43 and #32 are cleaner solutions. Closing these PRS.
